torch/futures/__init__.py

```diff
@@ -61,6 +61,15 @@ def then(self, callback):  # type: (Callable[[Future[T]], S]) -> Future[S]
             ``callback`` and will be marked as completed when the given
             ``callback`` finishes.
 
+        .. note:: Note that if the callback function throws, either
+            through the original future being completed with an exception and
+            calling ``fut.wait()``, or through other code in the callback, the
+            future returned by ``then`` will be marked appropriately with the
+            encountered error. However, if this callback later completes
```

Comment:
Probably better to mention this extra error handling is currently only needed by `add_done_callback()`, not `then()`?

Reply:
This can also happen with `then`, i.e. if users use `then` to schedule arbitrary callbacks. This is possible since `_add_done_callback` is currently a private api and not exposed. i.e., in the example in the PR, the same issue is possible with `then()`.